test/models/bot/alegre_3_test.rb

```diff
@@ -580,4 +580,40 @@ def self.extract_project_medias_from_context(search_result)
   test "should not resort matches if format is unknown" do
     assert_equal 'Foo', Bot::Alegre.return_prioritized_matches('Foo')
   end
+
+  test "should process webhook" do
```

Comment:
If I'm reading it correctly, looks like this test is not actually doing anything. I suggest you move this test to https://github.com/meedan/check-api/blob/develop/test/controllers/webhooks_controller_test.rb and change it to do the following:

1. Create testing data and stub Alegre response, like you're already doing
2. Be sure that the Redis key is nil (you can use `assert_nil`)
3. Call the webhook endpoint by executing `post :index, params: { name: :alegre }, body: payload` in the test (you'll need to define `payload` similarly to how you are defining `request` as an `OpenStruct` currently) - there are examples in https://github.com/meedan/check-api/blob/develop/test/controllers/webhooks_controller_test.rb
4. Be sure that the key in Redis is not null and contains the expected value (you can use `assert_no_nil` or even `assert_equal`)
5. Use `travel_to` and be sure that the key was removed from Redis